src/libs/clickHandler.js

```diff
@@ -5,82 +5,87 @@
-function getLinkTarget(target, props) {
-  return props.target || (target && target.target) || '_self';
-}
-
-function isNewWindow(target, props) {
-  return getLinkTarget(target, props) === '_blank';
+const isLink = (target) => target.tagName === 'A';
+const isButtonLike = (target) => {
+  const { tagName, type } = target;
+  if (tagName === 'BUTTON') {
+    return true;
+  }
+  // input with submit or button type
+  if (tagName === 'INPUT' && (type === 'submit' || type === 'button')) {
+    return true;
+  }
 }
 
-function isDefaultRedirectLink(target) {
-  let defaultRedirectLink = false;
+const isDefaultRedirectLink = (target) => {
   // if it's a
   // 1. link
   // 2. button
   // 3. input with submit or button type
   // then redirect it by default, otherwise
-  if (target.tagName === 'A' || target.tagName === 'BUTTON') {
-    defaultRedirectLink = true;
-  }
-  if (target.tagName === 'INPUT' && (target.type === 'submit' || target.type === 'button')) {
-    defaultRedirectLink = true;
+  const { tagName, type } = target;
+  if (isLink(target) || isButtonLike(target)) {
+    return true;
   }
-  return defaultRedirectLink;
+
+  return false;
 }
 
-function isFormSubmit(target) {
-  let formSubmit = false;
+const isFormSubmit = (target) => {
   // if it's a
   // 1. button
   // 2. input with submit or button type
-  if (
-    target.tagName === 'BUTTON'
-    || (target.tagName === 'INPUT' && (target.type === 'submit' || target.type === 'button'))
-  ) {
-    formSubmit = true;
+  if (isButtonLike(target)) {
+    return true;
   }
-  return formSubmit;
+  return false;
 }
 
 /**
  * clickHandler which integrate the beacon event, send out beacon event then redirect user to the destination.
  * @param {Object} e the click event
  * @method ClickHandler
  */
-function clickHandler(e) {
-  const self = this;
+const clickHandler = (e, options = {}) => {
   const target = e.target || e.srcElement;
+  const isForm = isFormSubmit(target);
+
   let isRedirectLink = isDefaultRedirectLink(target);
   let isPreventDefault = true;
 
-  const { props } = self;
-  const followLink = self._shouldFollowLink();
+  const {
+    executeEvent,
+    i13nNode,
+    props = {},
+    shouldFollowLink
+   } = options;
+
   let href = '';
 
-  // return and do nothing if the handler is append on a component without I13nMixin
-  if (!self.executeI13nEvent) {
+  if (!executeEvent) {
     return;
   }
 
+  const { follow } = props;
+
   href = props.href || target.href;
 
   // if users disable the redirect by follow, force set it as false
-  if (!isUndefined(followLink)) {
-    isRedirectLink = followLink;
-  }
+  isRedirectLink = (shouldFollowLink?.(props) ?? follow) ?? isRedirectLink;
 
-  // if it's not an anchor or this is a hash link url for page's internal links.
+  // 1. if it is an anchor but no href or hash link
+  // 2. button without form submit
   // Do not trigger navigate action. Let browser handle it natively.
-  if (!href || (href && href[0] === '#')) {
```

Comment:
this is actually an existing bug, when a button-ish type tag is used, we will never have `href`, and we always set the value to false even there is a `form`. the callback below will never be executed.